### PR TITLE
Swift: extract only primary files

### PR DESCRIPTION
**Pull request by AlexDenisov**
While the (removed) comment is correct and the frontend can be called in different modes, both `swift build` and `xcodebuild` always use `-primary-file` when compiling Swift projects.
The other mode was present only within our test runner (`qltest.sh`), so removing it and doing what the official build systems do simplifies our code base.

Additionally, file archival is now a separate function/operation.

### CODE REVIEW

File: swift/extractor/SwiftExtractor.cpp [hunk 1]
```diff
@@ -18,9 +18,7 @@
 
 using namespace codeql;
 
-static void extractFile(const SwiftExtractorConfiguration& config,
-                        swift::CompilerInstance& compiler,
-                        swift::SourceFile& file) {
+static void archiveFile(const SwiftExtractorConfiguration& config, swift::SourceFile* file) {
```
Comment (redsun82) [inline swift/extractor/SwiftExtractor.cpp:21]: nit: while the swift frontend library does use pointers, in our codebase I'd start using pointers only if we expect them to be nullable (I know, we don't do that yet in the dispatcher). I'd keep `swift::SourceFile` a reference (also for consistency with `config` and `compiler`).

File: swift/extractor/SwiftExtractor.cpp [hunk 2]
```diff
@@ -49,12 +47,16 @@ static void extractFile(const SwiftExtractorConfiguration& config,
               << dstFilePath.str().str() << "': " << ec.message() << "\n";
     return;
   }
+}
 
+static void extractFile(const SwiftExtractorConfiguration& config,
+                        swift::CompilerInstance& compiler,
+                        swift::SourceFile* file) {
```
Comment (redsun82) [inline swift/extractor/SwiftExtractor.cpp:54]: same here, I would rather use `swift::SourceFile&`

File: swift/extractor/SwiftExtractor.cpp (hunk 1 above)
Comment (AlexDenisov) [reply swift/extractor/SwiftExtractor.cpp:21]: Just for the record: are you suggesting to use references? 😆 

Ok, will switch back to them.

**Review by redsun82 — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**